common/djangoapps/third_party_auth/saml.py

```diff
@@ -112,10 +116,22 @@ def get_user_details(self, attributes):
         """
         details = super(EdXSAMLIdentityProvider, self).get_user_details(attributes)
         extra_field_definitions = self.conf.get('extra_field_definitions', [])
+        
         details.update({
             field['name']: attributes[field['urn']][0] if field['urn'] in attributes else None
             for field in extra_field_definitions
         })
+        
+        if configuration_helpers.get_value(
+                'ENABLE_REGISTRATION_USERNAME_SUGGESTION',
+                settings.FEATURES.get('ENABLE_REGISTRATION_USERNAME_SUGGESTION', False)):
+
+            username_generator_settings = self.conf.get('USERNAME_GENERATOR', {})
+            fullname = details['fullname']
```

Comment:
While testing this on the sandbox I realized that this will override the username, even if SAML provider already returns a value for username.

I think that instead of always using `details['fullname']` as the base for the generated username, we should prefer `details['username']`, if present. We can still pass the given username through the generator, to replace spaces/lowercase/append suffix if needed.

Additionally, we probably shouldn't expect `details['fullname']` to always be present.

Reply:
@mtyaka  we had previous discussions about it on our technical implementation doc. According to that document, we will receive (full name, ID called MOE, Institutions where the student learns and student age group ) from the identity provider. Therefore, we based the username suggestion on full name.

Reply:
@dfrojas Right, for this particular identity provider, the `username` field would not be present and `fullname` should be used as the base, but in order to make the code more generic and usable with other identity providers, I think it would make sense to use `username` *if it exists`. Bascially something like:

```python
username_base = details['username'] or details['fullname']
username = username_generator.generate_username(username_base)
```

Reply:
@mtyaka Get it, I just modified it